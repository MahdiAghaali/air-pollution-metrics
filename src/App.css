* {
  box-sizing: border-box;
}

:root {
  --blue: #4369b2;
  --halfBlue: #3f62a5;
  --darkBlue: #35548b;
  --lightBlue: #4a83e6;
}

html,
body {
  background-color: var(--blue);
  color: white;
  font-family: Arial, Helvetica, sans-serif;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100vh;
}
