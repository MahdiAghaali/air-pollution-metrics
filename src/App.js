import {
  BrowserRouter, Routes, Route,
} from 'react-router-dom';
import './App.css';
import { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import Header from './components/Header';
import ContinentList from './components/ContinentList';
import CountryList from './components/CountryList';
import Country from './components/Country';
import { getCountries } from './redux/countries/countries';
import { getAllCities } from './redux/countries/cities';

const App = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getCountries());
    dispatch(getAllCities());
  }, [dispatch]);
  return (
    <BrowserRouter>
      <Header />
      <Routes>
        <Route path="/" element={<ContinentList />} />
        <Route path="/continent/:continent" element={<CountryList />} />
        <Route path="/continent/:continent/country/:country" element={<Country />} />
      </Routes>
    </BrowserRouter>
  );
};

export default App;
