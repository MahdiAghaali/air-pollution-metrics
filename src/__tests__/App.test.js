import React from 'react';
import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import App from '../App';

afterEach(() => cleanup());

describe('Test App.js', () => {
  it('Test if the page loads', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <App />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
