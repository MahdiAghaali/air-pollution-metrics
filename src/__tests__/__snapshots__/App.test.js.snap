// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App.js Test if the page loads 1`] = `
Array [
  <header>
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        src="worldmap-mobile.png"
      />
    </a>
    <h1>
      Api Pollution Metrics
    </h1>
  </header>,
  <section
    id="continentList"
  >
    <h2>
      Start by continent
    </h2>
    <ul />
  </section>,
]
`;
